src/dash-table/components/Table/Table.less

```diff
@@ -334,6 +156,28 @@
             }
         }
 
+        &:not(.dash-empty-11) {
+            .row-0 {
+                tr:last-of-type {
+                    td, th {
+                        border-bottom: none !important;
+                    }
+                }
+            }
+        }
+
+        &:not(.dash-empty-01) {
+            .cell-0-0,
+            .cell-1-0 {
+                tr {
+                    td:last-of-type,
+                    th:last-of-type {
+                        border-right: none !important;
+                    }
+                }
+            }
+        }
```

Comment:
Handles the boundary cases for n_fixed_rows and n_fixed_columns -- only apply if the neighbouring table fragment is not empty